app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -814,15 +817,23 @@ void OpenXRInputSource::Update(const XrFrameState& frameState, XrSpace localSpac
     bool usingTrackedPointer = pointerMode == DeviceDelegate::PointerMode::TRACKED_POINTER;
     bool hasAim = isPoseActive && (poseLocation.locationFlags & XR_SPACE_LOCATION_ORIENTATION_VALID_BIT);
 
+
+    if (hasAim && mUsingHandInteractionProfile && handFacesHead) {
+        // XR_MSFT_hand_interaction returns true for aim when doing palm pinching. That's unexpected because
+        // as long as pinching finishes aim becomes false again. Force it to false to avoid showing
+        // the aim ray when palm pinching.
+        hasAim = false;
+    }
+
     // Don't enable aim for eye tracking as we don't want to paint the beam. Note that we'd still
     // have an aim, meaning that we can point, focus, click... UI elements.
     delegate.SetAimEnabled(mIndex, hasAim && usingTrackedPointer);
 
     // Disable the controller if there is no aim unless:
     // a) we're using hand interaction profile and we have hand tracking info. In that case the user
     // is facing their palm to do a hand gesture (that's why there is no aim)
-    // b) we're using eye tracking. In that case the eye gaze will be the aim, not the contorller's
-    if (!hasAim && !(mUsingHandInteractionProfile && gotHandTrackingInfo && handFacesHead) && !usingEyeTracking) {
```

Comment:
Why we don't need to check for handFracesHead anymore ?

Reply:
Because that condition was wrong, it was way too restrictive.

If we're using hand tracking and we have the joints data available (gotHandTrackingInfo==true) then we should still paint the hands even if they have no aim or they don't face the head. For example if
```
hasAim=false
mUsingHandInteractionProfile=true
gotHandTrackingInfo=true
handFacesHead=false
usingEyeTracking=false
```
resulted in a true condition that was disabling the controller (hand) when it should not. The previous condition was kind of assuming that having no aim meant that the hand was facing head, but that is not always correct, the tracking system might fail to provide aim in other cases.